content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -154,9 +154,130 @@ TODO.
 
 ## Extensionless asset support
 
-<div class="release-feature-authors">authors: @TODO</div>
+<div class="release-feature-authors">authors: @bushrat011899</div>
 
-TODO.
+In prior versions of Bevy, the default way to choose an [`AssetLoader`] for a particular asset was entirely based around file extensions. The recent addition of [`meta` files] allowed for specifying more granular loading behavior, but file extensions were still required. In Bevy 0.13, the asset type can now used to infer the [`AssetLoader`].
+
+```rust
+// Uses AudioSettingsAssetLoader
+let audio = asset_server.load("data/audio.json");
+
+// Uses GraphicsSettingsAssetLoader
+let graphics = asset_server.load("data/graphics.json");
+```
+
+This is possible because every [`AssetLoader`] is required to declare what **type** of asset it loads, not just the extensions it supports. Since the [`load`] method on [`AssetServer`] was already generic over the type of asset to return, this information is already available to the [`AssetServer`] so that the appropriate [`Handle`] type is returned.
+
+```rust
+// The above example with types shown
+let audio: Handle<AudioSettings> = asset_server.load::<AudioSettings>("data/audio.json");
+let graphics: Handle<GraphicsSettings> = asset_server.load::<GraphicsSettings>("data/graphics.json");
+```
+
+Now we can also use it to choose the [`AssetLoader`] itself.
+
+```rust
+#[derive(Resource)]
+struct Settings {
+    audio: Handle<AudioSettings>,
+//                ^^^^^^^^^^^^^
+//                | This type...
+}
+
+fn setup(mut settings: ResMut<Settings>, asset_server: Res<AssetServer>) {
+    settings.audio = asset_server.load("data/audio.json");
+//                                ^^^^
+//                                | ...is passed here...
+}
+
+impl AssetLoader for AudioSettingsAssetLoader {
+    type Asset = AudioSettings;
+//               ^^^^^^^^^^^^^
+//               | ...and checked against AssetLoader::Asset
+
+    /* snip */
+}
+```
+
+When loading an asset, the loader is chosen by checking (in order):
+
+1. The asset `meta` file
+2. The type of `Handle<A>` to return
+3. The file extension
+
+```rust
+// This will be inferred from context to be a Gltf asset, ignoring the file extension
+let gltf_handle = asset_server.load("models/cube/cube.gltf");
+
+// This still relies on file extension due to the label
```

Comment:
I presume the label mentioned is the `#Mesh0/Primitive0` ? Am unsure.

Reply:
Correct, I've updated this example to add a comment highlighting it as the cause of extension-based resolution.